### PR TITLE
check-ssl-cert: add `ca-file`, `cert-file`, `key-file`, `no-check-certificate` options

### PR DESCRIPTION
**Pull request by wafuwafu13**
Close: https://github.com/mackerelio/go-check-plugins/issues/392

### CODE REVIEW

**Comment by wafuwafu13** [comment]
ubuntu pass https://github.com/mackerelio/go-check-plugins/actions/runs/3927416006/jobs/6714019466
windows failed https://github.com/mackerelio/go-check-plugins/actions/runs/3927416006/jobs/6714019537
```zsh
-rw-r--r-- 1 runneradmin 197121 1706 Jan 16 05:16 ca.key
-rw-r--r-- 1 runneradmin 197121 1706 Jan 16 05:16 client.key
-rw-r--r-- 1 runneradmin 197121   27 Jan 16 05:16 extfile.txt
-rw-r--r-- 1 runneradmin 197121  945 Jan 16 05:16 gen-cert.sh
-rw-r--r-- 1 runneradmin 197121 1706 Jan 16 05:16 server.key
```

File: check-ssl-cert/lib/check-ssl-cert.go [hunk 1]
```diff
@@ -63,8 +67,29 @@ func run(args []string) *checkers.Checker {
...
+	tlsConfig := &tls.Config{}
+	if opts.CAfile != "" {
+		caCert, err := os.ReadFile(opts.CAfile)
+		if err != nil {
+			return nil, err
+		}
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
+		tlsConfig.RootCAs = caCertPool
+		tlsConfig.BuildNameToCertificate()
```
Comment (lufia) [inline check-ssl-cert/lib/check-ssl-cert.go:80]: [may] Today I noticed this method is marked as deprecated.
https://pkg.go.dev/crypto/tls#Config.BuildNameToCertificate

File: check-ssl-cert/lib/check-ssl-cert.go [hunk 2]
```diff
@@ -26,19 +30,19 @@ func parseArgs(args []string) (*certOpts, error) {
...
-	ckr := run(os.Args[1:])
+	ckr := Run(os.Args[1:])
 	ckr.Name = "SSL"
 	ckr.Exit()
 }
 
-func run(args []string) *checkers.Checker {
-	opts, err := parseArgs(os.Args[1:])
+func Run(args []string) *checkers.Checker {
+	opts, err := parseArgs(args)
```
Comment (lufia) [inline check-ssl-cert/lib/check-ssl-cert.go:39]: oops

File: check-ssl-cert/lib/check-ssl-cert_test.go [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+		"-duration",
+		"720h0m0s", // 30*24*time.Hour
+	)
+	cmd.Dir = tmpDir
+	err := cmd.Run()
+	if err != nil {
+		return "", err
+	}
+	os.Rename(filepath.Join(tmpDir, "cert.pem"), filepath.Join(tmpDir, "client.crt"))
+	os.Rename(filepath.Join(tmpDir, "key.pem"), filepath.Join(tmpDir, "client.key"))
```
Comment (wafuwafu13) [inline check-ssl-cert/lib/check-ssl-cert_test.go:45]: ref: https://github.com/golang/go/pull/57815

File: check-ssl-cert/lib/check-ssl-cert_test.go [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+)
+
+const (
+	cert_pem   = "cert.pem"
+	cert_key   = "key.pem"
+	client_crt = "client.crt"
+	client_key = "client.key"
+)
+
+// generate "cert.pem", "key.pem", "client.crt", "client.key" in tmpDir
```
Comment (lufia) [inline check-ssl-cert/lib/check-ssl-cert_test.go:27]: What did you think to generate locally those keys and certificates and to commit them into testdata/ directory?

Comment (lufia) [reply check-ssl-cert/lib/check-ssl-cert_test.go:27]: expiration.

**Review by lufia — APPROVED** [review]

**Pull request merged**